### ISSUE TITLE
Set active button via variables

### ISSUE THREAD

**Issue by notenoughtech** [AgP42/TaskerProjects#1]
Hi
Great work, would be nice to see the extended buttons controls so you could set the colours of the active button via a variable on load.

**Comment by AgP42** [comment]
Hello @notenoughtech 

I love your blog and youtube channel, so very glad that you appreciate my work too ;-)

I didn't understand exactly what you mean by "the extended buttons controls", but I understood that you would like to change the beautiful "orange" color that I choose for the selected mode. 
So I just updated the web screen and added a "color" configuration on the "Modes" categories of the screen variables to allow choosing another color for the active button. 

I hope it was your request (it was on my to-do list anyway...) !

Bye, 
AgP42

**Comment by notenoughtech** [comment]
Hi!
Thanks for the update!  
Just had a look at it! What I wanted to do is to pull information from my server with thermostat setting and use a %variable to set the active mode. let say I asked alexa to put the thermostat on HEAT - when I open tasker it will get all the info from the server and set the temp, ambient and also highlights the mode that the thermostat is already in
(I'm ok with a lovely orange!)

**Comment by AgP42** [comment]
Ah ok, then it is even easier because it is the default behavior, you just have to put the variable that represents the actual mode in the config "Modes/Actual Mode" and that's it. You just have to ensure it is exactly the same name.

Let's say your "Mode list" is :"Auto,Off,On" (no spaces !). 
Then you can have on the "Actual mode" : "%ActualMode". 
If %ActualMode" = "On", the web screen thermostat will display the mode "On" at startup. 

Personally I am using AutoRemote on my home automation system to "push" to Tasker any change of the mode or the target temperature. So anytime I open the web screen from any phone I am up to date !

Good for you ?

**Comment by notenoughtech** [comment]
I will check it out later on according to the instructions, I will have 2 addons to my system  - one with your webscreen *thank you and one as notification - here is the demo https://www.instagram.com/p/B3NGGMAj81U/

**Comment by AgP42** [comment]
Oh great !

For now with my web screen you can also have an update "pushed" by Tasker without the need to refresh the web screen manually. 
This update works for the temperature, the title and subtitle but not for the mode. But I can add it if you like !
So you can (already working) to do the same than on the video by asking Google or Alexa to set the heater to 21° for example, then with a Tasker profile you "catch" this new target and you request the display of the web screen again (the button "update" has to be toggle on to allow refresh only this value and not the full web screen)

Good luck and let me know if you have any other questions !

**Comment by notenoughtech** [comment]
I was thinking if I should resolve this via AR Join or HTTP.
I don't believe the dynamic updates would be needed (what are the chances) but i will have it in a persistent home notification with AN.

Since AN takes some HTML... if you want to take a crack at nice interface, I would not stop you from doing so ahhahah

**Comment by AgP42** [comment]
Yes AN takes some HTML, but it is really very limited and without any CSS it's not easy to design something nice... 

Personally I have done some widgets to display all the data (you can find here a tuto that I wrote about it : [/tutorial-display-dynamically-widgets-kwgt-or-zooper-according-to-autoremote-messages](https://forum.joaoapps.com/index.php?resources/tutorial-display-dynamically-widgets-kwgt-or-zooper-according-to-autoremote-messages.395/), here it's how it looks like:
![](https://i.ibb.co/pP8sjDY/main.jpg)
And when I press the button of one of the thermostat (the 8 top left widgets) it's open the web screen accordingly to this room. 

For each of my thermostat widget, I display the following infos : 
- temperature and humidity of the room (refresh by tasker every 15min if connected to the local wifi or every hour otherwise)
- mode and target temperature of the thermostat (updated by push AutoRemote)
- actual status of the heater (on/off) (updated by push AutoRemote)


If you like I can design a notification anyway, info to display : 
- actual temp or the room
- thermostat target and mode
- and a press on the notification itself to open the web screen thermostat

right ?

**Comment by AgP42** [comment]
Something like that : 
![](https://github.com/AgP42/TaskerProjects/blob/master/AutoTools%20Web%20Screen/thermostat/images/SM-N935F_20191006124454.gif?raw=true)

**Comment by notenoughtech** [comment]
That's pretty neat! :) You know you are doing all the hard work right now Agathe? You will leave me nothing to explain!  hahaa 👍

**Comment by AgP42** [comment]
I have made the hard work by designing the initial project, but just creating a notification and an action when all variables already exist, that's really easy !

Did you succeed to display the actual mode when you start the web screen ?

**Comment by notenoughtech** [comment]
I have the basics working, I will be implementing this into my system this
week and hope to post it on Friday.
I'm away for work so tinkering time is limited.

BTW you can already start thinking about multi-room Temp controls for the
WebScreen :)  My final plan is to connect each radiator via IoT valve so I
can control per room temperature. This way you can drive the heat where you
want, heating rooms quicker :)

Awesome work btw!

I appreciate the timely reploes

*Mat Zolnierczyk*
Owner: NotEnoughTECH
Editor: *pocketables.com <http://www.pocketables.com/about>**, *
*xda-developers.com* <https://www.xda-developers.com/author/notenoughtech/>
E: mat@notenoughtech.com
W: www.notenoughtech.com
<https://inbox.google.com/u/1/www.notenoughtech.com> | Twitter:
@notenoughtech <https://twitter.com/NotEnoughTECH>

**Comment by AgP42** [comment]
Hello Mat,

I already use it for 8 rooms with the following config : 

<a href="https://ibb.co/WgFz571"><img src="https://i.ibb.co/WgFz571/Screenshot-20191007-184948-Tasker.jpg" alt="Screenshot-20191007-184948-Tasker" border="0"></a>

To call the thermostat I use an AutoApps command "displaythermostatfor=:=2" (requested by a click on the corresponding widget), then I catch the "2" on the variable "%room". And I have several arrays, all on the same order to represent the value of each room. So the hard ("long" more that "hard"...) work is on the definition of the arrays, but then I can display as much thermostat as I want with only 1 task "web screen" ! And once the structure is done, I can easily add a room. 
<a href="https://ibb.co/Jc1q6ct"><img src="https://i.ibb.co/Jc1q6ct/Screenshot-20191007-185838-Tasker.jpg" alt="Screenshot-20191007-185838-Tasker" border="0"></a>
(The number represent the variable associated on my home automation system)

You can do the same kind of behavior with a persistent "button notification" with 1 button for each heater, each button will send a different "displaythermostatfor=:=x" to display the web screen with the corresponding room values. Something like that, no? : 
![](https://github.com/AgP42/TaskerProjects/blob/master/AutoTools%20Web%20Screen/thermostat/images/SM-N935F_20191007193541.gif?raw=true)

What was the "multiroom" behavior that you had in mind ?

I also have to go to work every day, then second day taking care of my baby (also a "DIY project", with no bug at the first try! ;-) my best engineering work ever!), but finishing a day playing with Tasker is a very pleasant hobby for the evening ;-) 
My pleasure to share my work with you !

Bye, 
Agathe
